apps/myaccount/src/components/shared/system-notification-alert.tsx

```diff
@@ -0,0 +1,207 @@
+                        )
+                    }));
+                }
+            })
+            .catch((error: AxiosError) => {
+                if (error?.response?.data?.detail) {
+                    dispatch(addAlert({
+                        description: t(
+                            "myAccount:components.selfSignUp.preference.notifications.error.description",
+                            { description: error.response.data.detail }
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t("myAccount:components.selfSignUp.preference.notifications.error.message")
+                    }));
+
+                    return;
+                }
+
+                dispatch(addAlert({
+                    description: t(
+                        "myAccount:components.selfSignUp.preference.notifications.genericError.description"
+                    ),
+                    level: AlertLevels.ERROR,
+                    message: t("myAccount:components.selfSignUp.preference.notifications.genericError.message")
+                }));
+            });
+    };
+
+    /**
+     * Resends the account confirmation Email link.
+     */
+    const resendAccountConfirmationLink = (): void => {
+        if (isSubmitting) {
+            return;
+        }
+        setIsSubmitting(true);
+        resendVerificationLinkOrCode(RecoveryScenario.SELF_SIGN_UP)
+            .then(() => {
+                dispatch(addAlert({
+                    description: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendSuccess.description"),
+                    level: AlertLevels.SUCCESS,
+                    message: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendSuccess.message")
+                }));
+            })
+            .catch((errorMessage: AxiosError) => {
+                dispatch(addAlert({
+                    description: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendError.description", { error: errorMessage }),
+                    level: AlertLevels.ERROR,
+                    message: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendError.message")
+                }));
+            }).finally(() => {
+                setIsSubmitting(false);
+            });
+    };
+
+    /*
+    * The resending capability is available only when account confirmation is configured
+    * to use an Email Link. The rationale for excluding Email OTP and SMS OTP confirmation
+    *  methods is that their respective code validations must occur during the initial
+    * user registration flow itself rather in a later stage via MyAccount.
+    */
+    if (isAccountStatePendingSR && isPreferredChannelEmail && !isSRSendOtpInEmailEnabled) {
```

Comment:
let's take the condition and message as props so that system notification alert component can be reused for other notification scenarios as well.